src/Resources/skeleton/registration/RegistrationController.tpl.php

```diff
@@ -58,4 +87,29 @@ public function register(Request $request, UserPasswordEncoderInterface $passwor
             'registrationForm' => $form->createView(),
         ]);
     }
+<?php if ($will_verify_email): ?>
+
+    /**
+     * @Route("/verify", name="app_validate_user")
+     */
+    public function verifyUserEmail(Request $request): Response
+    {
+        $this->denyAccessUnlessGranted('IS_AUTHENTICATED_FULLY');
+
+        // validate email confirmation link - sets User::verified=true and persists
+        // if confirmation fails, an exception is thrown that extends VerifyEmailExceptionInterface
+        try {
+            $this->verifier->handleEmailConfirmation($request, $this->getUser());
+        } catch (VerifyEmailExceptionInterface $exception) {
+            $this->addFlash('verify_email_error', $exception->getReason());
+
+            return $this->redirectToRoute('app_register');
```

Comment:
should use `$route_name` for this - I'm not sure it's dynamic, but we *are* passing in a variable

Reply:
done